From a7a64988dc87f09946461508c347662a1ee85e02 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sat, 9 Dec 2017 20:50:45 +0100
Subject: [PATCH 6/6] Ensure signal.h is included

---
 include/ma_pthread.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/ma_pthread.h b/include/ma_pthread.h
index 28a0a8c..12baf5a 100644
--- a/include/ma_pthread.h
+++ b/include/ma_pthread.h
@@ -208,6 +208,7 @@ extern int my_sigwait(const sigset_t *set,int *sig);
 #endif
 #undef _REENTRANT			/* Fix if _REENTRANT is in pthread.h */
 #include <pthread.h>
+#include <signal.h>
 #ifndef _REENTRANT
 #define _REENTRANT
 #endif
-- 
2.15.1

